### PR TITLE
Add rateLimit event

### PR DESCRIPTION
**Pull request by TheFloppyBanana**
Allow monitoring of your app's ratelimits in your logs whenever it hits one.

### CODE REVIEW

**Comment by bdistin** [comment]
The log level wouldn't be wtf, because the dev hasn't done anything wrong. Should be debug

**Comment by TheFloppyBanana** [comment]
Thank you for the heads-up! I've pushed a change regarding this. 😄

File: events/rateLimit.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const { Event } = require('klasa');
+
+const HEADER = `\x1b[31m[RATELIMIT]\x1b[0m`;
```
Comment (kyranet) [inline events/rateLimit.js]: This should be using the Colors class

File: events/rateLimit.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+module.exports = class extends Event {
+
+	run({ timeout, limit, method, path, route }) {
+		this.client.emit('verbose',
+			`${HEADER} \nTimeout: ${timeout}ms\nLimit: ${limit} requests\nMethod: ${method.toUpperCase()}\nPath ${path}\nRoute: ${route}`);
```
Comment (kyranet) [inline events/rateLimit.js]: 1. Can me shortened using an array (I'm pretty sure they'll be joined by newline).
1. Path and route are identical, use one of them.

File: events/rateLimit.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/**
+ * KlasaConsoleConfig#useColor must be
+ * enabled in order to use this piece.
+ */
+const { Event, Colors } = require('klasa');
+
+const HEADER = new Colors({ text: 'red' });
```
Comment (kyranet) [inline events/rateLimit.js]: You can also do

```javascript
const HEADER = new Colors({ text: 'red' }).format('[RATELIMIT]');
```

And then inside the method, put `HEADER`, so you don't format every time you hit ratelimits.

Comment (TheFloppyBanana) [reply events/rateLimit.js]: Thank you for the suggestion! I'm gonna push a change right away. :3

**Pull request merged**